src/components/account/GetAccountId.tsx

```diff
@@ -0,0 +1,79 @@
+import RequestStatus from 'components/RequestStatus'
+import React, { useEffect, useState } from 'react'
+import validateAccountId from 'utils/validateAccountId'
+
+const GetAccountId = () => {
+  const [accountId, setAccountId] = useState('')
+  const [accountIdResult, setAccountIdResult] =
+    useState<AccountResponse | null>()
+
+  const [requestStatus, setRequestStatus] =
+    useState<RequestStatusOptions>('idle')
+
+  const [errorMessage, setErrorMessage] = useState<string | null>(null)
+
+  useEffect(() => {
+    if (requestStatus !== 'success') setAccountIdResult(null)
+  }, [requestStatus])
+
+  return (
+    <div className="sectionContainer">
+      <h3>Search for an account</h3>
+      <input
+        type="text"
+        value={accountId}
+        placeholder="9e28507a-632f-44a6-99a7-98695cf2adcf"
+        onChange={(e) => setAccountId(e.target.value)}
+      ></input>
+      <button
+        className="getAccountIdButton"
+        onClick={() => {
+          const validateInput = validateAccountId(accountId)
+          if (validateInput) {
+            setErrorMessage(validateInput)
+            setRequestStatus('error')
+          } else {
+            setRequestStatus('fetching')
+            fetch(`https://nestjs-bank-app.herokuapp.com/accounts/${accountId}`)
+              .then((res) => {
+                if (!res.ok) {
+                  setRequestStatus('error')
+                  throw Error(res.statusText)
+                }
+                setRequestStatus('success')
+                return res.json()
+              })
+              .then((data) => setAccountIdResult(data))
+              .catch((e) => {
+                setErrorMessage(e.message)
+              })
+          }
+        }}
+      >
+        Search
+      </button>
+
+      <div className="requestStatus">
+        {requestStatus === 'fetching' || requestStatus === 'error' ? (
+          <RequestStatus requestStatus={requestStatus} message={errorMessage} />
+        ) : null}
+      </div>
+      {accountIdResult ? (
```

Comment:
typescript is smart to narrow your types by checking for their truthiness. Inside the truth block of this ternary, typescript knows that an account response object exists and will narrow the type for you. With this in mind, you don't have to perform to optional chaining to check if a key exists. You can test this by hovering over the object in the ternary.

Here we know`accountIdResult` can be truthy, undefined, or null
![Screen Shot 2021-12-28 at 2 11 21 PM](https://user-images.githubusercontent.com/39835725/147607405-dc609ee4-287f-411c-8689-d62889cce25a.png)

Inside the truth block, we know an`accountIdResult` exists
![Screen Shot 2021-12-28 at 2 11 49 PM](https://user-images.githubusercontent.com/39835725/147607433-2daf6bf5-1314-4b47-a814-86e7dcb40077.png)

[More on type narrowing ](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#truthiness-narrowing)